src/global/common.rs

```diff
@@ -478,6 +480,29 @@ pub(crate) async fn get_expose_scripts_sync_status(
     Ok((to_remove, to_add))
 }
 
+/// Check if they were all auto-exposed, or if the user manually exposed a subset of them
+pub fn check_auto_exposed(
+    env_binaries: &IndexMap<PackageName, Vec<(String, PathBuf)>>,
+    exposed_mapping_binaries: &IndexSet<Mapping>,
+) -> bool {
+    let env_binaries_names = env_binaries
+        .values()
+        .flatten()
+        .map(|(name, _)| name)
+        .collect_vec();
+
+    let exposed_binaries_names = exposed_mapping_binaries
+        .iter()
+        .map(|mapping| mapping.executable_name())
+        .collect_vec();
```

Comment:
Same here

Reply:
^ here I think it will not work because I need to check if a name is in this exposed list. Changed it to hashset though ( instead of vec )